### PR TITLE
[14.0][IMP][FWD] account_financial_report: extensibility: ml fields 

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Forward port of https://github.com/OCA/account-financial-reporting/pull/951

cc @ForgeFlow

### CODE REVIEW

**Review by DavidJForgeFlow — APPROVED** [review]
LGTM

**Review by mariadforgeflow — APPROVED** [review]
LGTM!

**Review by JoanSForgeFlow — APPROVED** [review]

File: account_financial_report/report/abstract_report_xlsx.py [hunk 1]
```diff
@@ -297,7 +297,9 @@ def write_line_from_dict(self, line_dict, report_data):
                     value or "",
                     report_data["formats"]["format_right"],
                 )
-        report_data["row_pos"] += 1
+            else:
+                self.write_non_standard_column(cell_type, col_pos, value)
+        self.row_pos += 1
```
Comment (pedrobaeza) [inline account_financial_report/report/abstract_report_xlsx.py]: Why changing this? It seems incorrect

Comment (AaronHForgeFlow) [reply account_financial_report/report/abstract_report_xlsx.py]: In my use case. I am adding the purchase order information to the report. When I query the purchase_id field in the method "_get_report_columns" I do it this way:

```

            6: {
                "header": _("Purchase Order"),
                "field": "purchase_id",
                "width": 11,
                "type": "purchase",
            },
```

With that hook I can then write the value I need. Putting type "many2one" in the columns declaration was not working in my case (at least in v13). Then I decided to put the hook so I can take the value when printing the line. I will check in this version if it works without the hook.

Comment (pedrobaeza) [reply account_financial_report/report/abstract_report_xlsx.py]: I was just talking about changing `report_data["row_pos"] += 1` with `self.row_pos += 1`

Comment (AaronHForgeFlow) [reply account_financial_report/report/abstract_report_xlsx.py]: I overlooked this. Yes, that is non intentional and seems an incorrect merge conflict resolution.

Comment (AaronHForgeFlow) [reply account_financial_report/report/abstract_report_xlsx.py]: Fixed and it is working now. Sorry all.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**